src/common/storeKey.ts

```diff
@@ -0,0 +1,28 @@
+export interface IStoreKey<T> {
+  name: string;
+}
+
+/**
+ * Store keys are a very simple abstraction that maps a string name to an interface type.
+ * The enables a type-safe contract for store dependencies when you connect dumb components
+ * to stores.
+ *
+ * @example
+ * let selectionKey = storeKey<ISelection>('selection');
+ *
+ * let stores = new StoreSet()
+ *   .add(selectionKey, new Selection());
+ *
+ * <StoreHost stores={ stores }>
+ *   ...
+ * </StoreHost>
+ *
+ * connect(DumbComponent, [ selectionKey ], (props, selection) => ({
+ *   isSelected: selection.getSelected(props.item.key)
+ * });
+ *
+ * Note that the "selection" reference in the connect callback would be of type ISelection due
+ * to the selectionKey reference, and you will get TypeScript errors by referring to selection
+ * member that aren't in that interface.
+ */
+export const storeKey = <T>(name: string) => ({ name }) as IStoreKey<T>;
```

Comment:
A reason to use a class instead is to enforce that apps do not just define their own `IStoreKey<T>` instances and instead always use `new StoreKey<T>`. You can then do this to make it obvious:

``` typescript
export class StoreKey<T> {
    protected __StoreKey_Brand: void;

    public id: string;

    public name: string;

    constructor(name: string) {
        this.id = `${++lastId}`;
        this.name = name;
    }
}
```